packages/headless/src/controllers/commerce/context/cart/headless-cart.ts

```diff
@@ -90,7 +84,7 @@ export interface Cart extends Controller {
    *
    * @param item - The cart item to create, update, or delete.
    */
-  updateItem(item: CartItem): void;
+  updateItemQuantity(item: CartItem): void;
```

Comment:
As we use `name` in the key, updating the name causes a create operation whilst leaving the existing product in the cart. This extends to anything we add in the key, including price. Keeping the key as simple as possible address's this issue.

Reply:
The price actually has the same issue. We can therefore only update quantity, without doing a silent create operation.

Reply:
Ah ok, I think I have a better understanding. I think it means we have to change the JSDOC here still, especially on line 75-76

> - Setting `quantity` to a positive number updates the item's `name`, `price`, and `quantity` to the
   * specified values.
   
From my understanding, we don't really update the item's name & price, because the combination of all attributes represent a single item.